workers/unity/Assets/Gdk/Android/Utility/DeviceInfo.cs

```diff
@@ -0,0 +1,29 @@
+using UnityEngine;
+
+namespace Improbable.Gdk.Android
+{
+    public class DeviceInfo
+    {
+        private static readonly AndroidJavaObject Build = new AndroidJavaObject("android.os.Build");
+        private static readonly string Fingerprint = GetValue("FINGERPRINT");
+        private static readonly string Model = GetValue("MODEL");
+        private static readonly string Manufacturer = GetValue("MANUFACTURER");
+        private static readonly string Brand = GetValue("BRAND");
+        private static readonly string Device = GetValue("DEVICE");
+        private static readonly string Product = GetValue("PRODUCT");
+
+        public static bool IsEmulator()
+        {
+            return Fingerprint.StartsWith("generic") || Fingerprint.StartsWith("unknown") ||
+                Model.Contains("google_sdk") || Model.Contains("Emulator") ||
+                Model.Contains("Android SDK built for x86") || Manufacturer.Contains("Genymotion") ||
```

Comment:
`Android SDK built for x86`: you can create emulators for arm. does that also check for them?

Reply:
Resolved offline: This should theoretically cover ARM devices as well but it is really difficult to test (as computers are on x86 architecture and ARM emulators are extremely slow). If we get feedback in future that this is a required feature and it does not work we can look into it. Should be fine though.